### DISCUSSION TITLE
Random-accessing via GraphQL/REST?

### DISCUSSION THREAD

**Discussion by KisaragiEffective** [community/community#24736] [Apps, API and Webhooks]
<p>Hi, I’ve been exploring it for a while. That’s my query:</p>
<pre><code class="lang-auto">query { 
  repository(owner: "somewhere", name: "repo") {
  	pullRequests(first: 100, states: MERGED) {
      edges {
        node {
          number
          author {
            login
          }
          additions
          deletions
        }
      }
    }
  }
}
</code></pre>
<p>However, I couldn’t found a method to access intermidate data, and didn’t see like <code>page</code> or <code>offset</code>. Is it possible on GraphQL? How about on REST API?</p>

**Comment by dennwc** [comment, marked as answer]
<p>Try this query:</p>
<pre><code class="lang-auto">query { 
  repository(owner: "somewhere", name: "repo") {
  	pullRequests(first: 100, states: MERGED) {
      totalCount
      pageInfo { endCursor, hasNextPage }
      nodes {
          number
          author {
            login
          }
          additions
          deletions
       }
    }
  }
}
</code></pre>
<p>You can then use <code>endCursor</code> to query the next page.</p>

**Comment by KisaragiEffective** [comment]
<p>Okay, I’ve found out that <code>pullRequests(first: 100, states: MERGED, after: "cursor_here")</code> does work, where <code>"cursor_here"</code> is <code>endCursor</code> in the previous query.<br>
Thank you for your answer!</p>
